src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -205,6 +205,14 @@ private void SetVerb()
                         SetCurlOption(CURLoption.CURLOPT_COPYPOSTFIELDS, string.Empty);
                     }
                 }
+                else
+                {
+                    SetCurlOption(CURLoption.CURLOPT_CUSTOMREQUEST, _requestMessage.Method.Method);
+                    if (_requestMessage.Content != null)
+                    {
+                        SetCurlOption(CURLoption.CURLOPT_UPLOAD, 1L);
```

Comment:
> From David's response, sounds like at least TRACE should be special cased not to send it,
Yeah. I will send a new iteration for TRACE.

For DELETE and OPTIONS, content is allowed. There are two tests for these with content as well.